json_rpc/servers/websocketserver.nim

```diff
@@ -10,18 +10,51 @@ logScope:
   topics = "JSONRPC-WS-SERVER"
 
 type
+  RpcWebSocketServerAuth* = ##\
+    ## Authenticator function. On error, the resulting `HttpCode` is sent back\
+    ## to the client and the `string` argument will be used in an exception,\
+    ## following.
+    proc(req: HttpTable): Result[void,(HttpCode,string)]
+      {.gcsafe, raises: [Defect].}
+
   RpcWebSocketServer* = ref object of RpcServer
-    authHook: seq[Hook]
+    authHook: Option[RpcWebSocketServerAuth] ## Authorization call back handler
     server: StreamServer
     wsserver: WSServer
 
+  HookEx = ref object of Hook
+    handler: RpcWebSocketServerAuth ## from `RpcWebSocketServer`
+    request: HttpRequest            ## current request needed for error response
+
+proc authWithHtCodeResponse(ctx: Hook, headers: HttpTable):
+            Future[Result[void, string]] {.async, gcsafe, raises: [Defect].} =
```

Comment:
async functions should not have direct raises annotations - gcsafe should also not be needed, why is it here?